### DISCUSSION TITLE
Optimising RAVE on M1 Macbook Neural Chip

### DISCUSSION THREAD

**Discussion by umaximu** [acids-ircam/RAVE#298] [Q&A]
Hi, this is my first post on Github, and my first project using any kind of coding as a musician. I am so excited to use this technology, it's like a revelation! I have tried running a few of the available RAVE Colabs but encountered some issues, and after splashing a bit of cash on the A100 GPU to little result I decided to just dive in and run RAVE locally on my M1 Macbook Air (boo, I know). It's obviously quite slow, and I'm happy to bear with it if I must, but at the moment it looks like it will take about 40 days to run 1,000,000 steps, and that seems a little crummy.

I saw that Apple has recently announced MPS Backend support for PyTorch, and I was wondering if this could be used to speed up the process by running RAVE on the Neural Engine? Has anyone had any luck with this, and if so how did you go about it?

Thanks in advance!

**Comment by scztt** [comment]
At least for playback, I was able to use the most recent PyTorch nightlies with GPU mode enabled - the audio seems fine and consistent with CPU mode, though I haven't tested closely.
